cmd/rebuilddb2/rebuilddb2.go

```diff
@@ -243,28 +306,55 @@ func mainCore() error {
 	speedReport()
 
 	if reindexing || cfg.ForceReindex {
+		if err = db.DeleteDuplicates(); err != nil {
+			return err
+		}
+
+		// Create indexes
 		if err = db.IndexAll(); err != nil {
 			return fmt.Errorf("IndexAll failed: %v", err)
 		}
-		if !cfg.UpdateAddrSpendInfo {
+		// Only reindex address table here if we do not do it below
+		if cfg.AddrSpendInfoOnline {
 			err = db.IndexAddressTable()
 		}
+		if !cfg.TicketSpendInfoBatch {
+			err = db.IndexTicketsTable()
+		}
 	}
 
-	if cfg.UpdateAddrSpendInfo {
-		// Remove existing indexes not on funding txns
+	if !cfg.AddrSpendInfoOnline {
+		// Remove indexes not on funding txns (remove on address table indexes)
 		_ = db.DeindexAddressTable() // ignore errors for non-existent indexes
+		db.EnableDuplicateCheckOnInsert(false)
 		log.Infof("Populating spending tx info in address table...")
 		numAddresses, err := db.UpdateSpendingInfoInAllAddresses()
 		if err != nil {
 			log.Errorf("UpdateSpendingInfoInAllAddresses FAILED: %v", err)
 		}
+		// Index address table
 		log.Infof("Updated %d rows of address table", numAddresses)
 		if err = db.IndexAddressTable(); err != nil {
 			log.Errorf("IndexAddressTable FAILED: %v", err)
 		}
 	}
 
+	if cfg.TicketSpendInfoBatch {
```

Comment:
Unlike the address table where updating it was faster in batch at the end, the tickets table can be updated efficiently online/during sync.